tracking_manager/models/ir_model.py

```diff
@@ -0,0 +1,205 @@
+                ).mapped("name")
+            res[model.model] = {"fields": tracked_fields, "notify": []}
+
+        for field in fields:
+            model_name = field.model_id.model
+            if (
+                model_name in self.env
+                and self.env[model_name]._fields.get(field.name)
+                and field.relation in res
+            ):
+                res[field.relation]["notify"].append(
+                    [self.env[model_name]._fields[field.name].inverse_name, field.name]
+                )
+        return res
+
+    @api.depends("active_custom_tracking")
+    def _compute_automatic_custom_tracking(self):
+        for record in self:
+            record.automatic_custom_tracking = False
+
+    def _default_automatic_custom_tracking_domain_rules(self):
+        return {
+            "product.product": [
+                "|",
+                ("ttype", "!=", "one2many"),
+                ("name", "in", ["barcode_ids"]),
+            ],
+            "sale.order": [
+                "|",
+                ("ttype", "!=", "one2many"),
+                ("name", "in", ["order_line"]),
+            ],
+            "account.move": [
+                "|",
+                ("ttype", "!=", "one2many"),
+                ("name", "in", ["invoice_line_ids"]),
+            ],
+            "default_automatic_rule": [("ttype", "!=", "one2many")],
+        }
+
+    @api.depends("automatic_custom_tracking")
+    def _compute_automatic_custom_tracking_domain(self):
+        rules = self._default_automatic_custom_tracking_domain_rules()
+        for record in self:
+            record.automatic_custom_tracking_domain = str(
+                rules.get(record.model) or rules.get("default_automatic_rule")
+            )
+
+    def update_custom_tracking(self):
+        for record in self:
+            fields = record.field_id.filtered("trackable").filtered_domain(
+                literal_eval(record.automatic_custom_tracking_domain)
+            )
+            fields.write({"custom_tracking": True})
+            untrack_fields = record.field_id - fields
+            untrack_fields.write({"custom_tracking": False})
+
+    @api.depends("field_id.custom_tracking")
+    def _compute_tracked_field_count(self):
+        for rec in self:
+            rec.tracked_field_count = len(rec.field_id.filtered("custom_tracking"))
+
+    def write(self, vals):
+        if "active_custom_tracking" in vals:
+            self.clear_caches()
+        return super().write(vals)
+
+
+class IrModelFields(models.Model):
+    _inherit = "ir.model.fields"
+
+    custom_tracking = fields.Boolean(
+        compute="_compute_custom_tracking",
+        store=True,
+        readonly=False,
+    )
+    native_tracking = fields.Boolean(
+        compute="_compute_native_tracking",
+        store=True,
+    )
+    trackable = fields.Boolean(
+        compute="_compute_trackable",
+        store=True,
+    )
+
+    @api.depends("native_tracking")
```

Comment:
```suggestion
    @api.depends("native_tracking", "model_id", "model_id.automatic_custom_tracking", "model_id.automatic_custom_tracking_domain")
```
to be sure the field is properly recomputed

Reply:
Hi dam, it's wanted, let's explain

The field automatic_custom_tracking is a computed readonly=False field.

So the idea is to only update field in some case keep existing value in most of the case.


Case 1: You install a new module that add some extra field to an existing model => if the field is track natively or match the domain, it's should be tracked.

Case 2: you activate or desactivate the option automatic_custom_tracking on the model => we do nothing as user can tick and untick and we do not want to reset all existing configuration.

Case 3: you modify the default domain (used as rule when a module install new field) => we do nothing as user can update the domain for new installed field but we do not want do change all existing configuration.

If a user want to reapply the domain on all fields then he can click on the button "update_custom_tracking" that we reset the value of all field https://github.com/OCA/server-tools/pull/2434/files/233e08cef4e959f7754f93e5ce0b3c6a3c84c732#diff-d2127356ce28776a51b642c4bf0e6c7762702ac27abc335c3b9eb9c204cb0d32R129

Thanks for your feedback